ReadMe.md

```diff
@@ -1,96 +1,141 @@
+
+### 1., 2. and 3: Refine Task Service, Unit testing and test design
+- We made sure to refine our code, and made unit tests for our validator part. We also made sure to use Equivalence Partitioning and Boundary Value Analysis for our tests.
+Our unit tests are made based on this data:
+- Description length:
+  - Valid, too short, too long ( minimum 5, maximum 255)
+  - Boundary value: 4-5, 255-256
+- Category length:
+  - Valid, too short, too long ( minimum 2, maximum 50)
+  - Boundary value: 1-2, 50-51
+- Deadline:
+  - Valid, invalid ( minimum datetime now -1 min, maximum datetime is as long as the calendar can show)
+  - Boundary value: now to 1 min ago, now to 1 min in the future
+
+In our Validator.cs file and ValidatorTest.cs, the validator is tested with the above data.
+
+### 4. Static Code Analysis (PMD or FxCop)
+- Integrate **PMD** (Java) or **FxCop** (C#) into your build process.
+- Fix at least **three critical issues** identified by the static code analyzer.
+- Submit the report and the corrected code.
+
+### 5. Code Coverage (JaCoCo or Coverlet)
+- Use **JaCoCo** (Java) or **Coverlet** (C#) for code coverage.
+- Ensure at least **75% code coverage**.
+- Submit the coverage report.
+
+### 6. Optional: Code Quality with SonarQube
+- If possible, integrate **SonarQube** for deeper code analysis.
+- Address and fix at least **three major issues** detected.
+- Submit the SonarQube report (optional).
+
+### 7. Peer Code Review
+- Conduct a **code review** with a peer based on principles from the [Code Review](https://en.wikipedia.org/wiki/Code_review) article.
+- Focus on:
+    - Code readability and maintainability.
+    - Adherence to coding standards.
+    - Correct use of mocks in tests.
+    - Identification of potential bugs.
+- Submit written feedback and any suggested code changes.
+
+We always make code review on everyones code. We use github pull requests to review the code, and below are some examples of some reviews we did:
+
+![Code comment 1](Images/codeComment1.png)
+
+![Code comment 2](Images/codeComment2.png)
+
+Especially in this last comment, we can see that no one is 'in charge' of the code, and that we are always working together to make sure what works, what doesnt and so on.
+We always check the code for readability and maintainability, and make sure that the code is up to our standards.
+
+
+### 8. Software Review
+- Perform a **Software Review** based on the [Software Review](https://en.wikipedia.org/wiki/Software_review) article:
+    - Functional correctness: Does the code do what it is supposed to do?
+    - Non-functional aspects (performance, maintainability, etc.).
+    - Best practices (code structure, documentation).
+- Submit a **300-word report** reflecting on your findings and improvements.
+
+In relation to functional correctness, the code performs as expected and passes all the test cases. Testing was done using equivalence partitioning and boundary value analysis. 
```

Comment:
Vi mangler at få specificeret hvor vi bruger equivalence partitioning & boundy value analysis.

for det er kun i vores validation tests det bliver anvendt / unit tests. det bliver ikke brugt i vores integration_api_tests som reviewers nok ville kommentere på modsiger hvad vi skriver her